packages/bot-web-ui/src/utils/__tests__/journal-notifications.spec.tsx

```diff
@@ -0,0 +1,132 @@
+import { messageWithButton } from 'Components/notify-item';
+import { isCustomJournalMessage } from '../journal-notifications';
+import { localize } from '@deriv/translations';
+
+describe('isCustomJournalMessage', () => {
+    const centerAndHighlightBlock = jest.fn();
+    const showErrorMessage = jest.fn(() =>
+        render(
+            messageWithButton({
+                unique_id: 1,
+                type: 'error',
+                message: '',
+                btn_text: localize('Go to block'),
+                onClick: () => {
+                    centerAndHighlightBlock();
+                },
+            })
+        )
+    );
+
+    const pushMessage = jest.fn();
+
+    beforeEach(() => {
+        showErrorMessage.mockClear();
+        centerAndHighlightBlock.mockClear();
+        pushMessage.mockClear();
+    });
+
+    it('should show error message with button when message is undefined and variable_name is not null', () => {
+        const message = {
+            message: undefined,
+            block_id: 123,
+            variable_name: 'testVariable',
+        };
+
+        isCustomJournalMessage(message, showErrorMessage, centerAndHighlightBlock, pushMessage);
+
+        expect(showErrorMessage).toHaveBeenCalledWith(expect.any(Object));
+    });
+
+    it('should show error message with button when message is not undefined and variable_name is null', () => {
+        const message = {
+            message: [['item1']],
+            block_id: 123,
+            variable_name: 'null',
+        };
+
+        isCustomJournalMessage(message, showErrorMessage, centerAndHighlightBlock, pushMessage);
+
+        setTimeout(() => {
+            expect(pushMessage).toHaveBeenCalledWith(expect.any(Object));
+        });
+    });
+
+    it('should push "NULL" message when message is null', () => {
+        const message = {
+            message: null,
+            block_id: 123,
+            variable_name: 'testVariable',
+        };
+
+        isCustomJournalMessage(message, showErrorMessage, centerAndHighlightBlock, pushMessage);
+
+        expect(pushMessage).toHaveBeenCalledWith('NULL');
+    });
+
+    it('should show error message with button when message is NaN', () => {
+        const message = {
+            message: NaN,
+            block_id: 123,
+            variable_name: 'testVariable',
+        };
+
+        isCustomJournalMessage(message, showErrorMessage, centerAndHighlightBlock, pushMessage);
+
+        expect(showErrorMessage).toHaveBeenCalledWith(expect.any(Object));
+    });
+
+    it('should push array as message when message is an array', () => {
+        const message = {
+            message: [['item1']],
+            block_id: 123,
+            variable_name: 'testVariable',
+        };
+
+        isCustomJournalMessage(message, showErrorMessage, centerAndHighlightBlock, pushMessage);
+
+        setTimeout(() => {
+            expect(pushMessage).toHaveBeenCalledWith(expect.any(Object));
+        });
```

Comment:
why do we need setTimeout @maryia-matskevich-deriv ?

Reply:
to make a delay, without a delay the test breaks down

Reply:
Good point @rupato-deriv but using **async/await** with **waitFor** or **act** from `@testing-library/react` test case still fails for some reason... so I kept  `setTimeout()`. If you find a better solution to this place, let me know. Maybe I haven't tried all possible combinations

Reply:
changed✅😌